src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLabels.scala

```diff
@@ -1,17 +1,27 @@
 package mesosphere.marathon.core.launcher.impl
 
 import mesosphere.marathon.core.task.Task
+import org.apache.mesos.Protos.Label
 import org.apache.mesos.{ Protos => MesosProtos }
 
+import scala.collection.mutable
+
 object TaskLabels {
   private[this] final val TASK_ID_LABEL = "marathon_task_id"
 
+  def taskIdForResource(resource: MesosProtos.Resource): Option[Task.Id] = {
```

Comment:
Do you think it makes sense to also filter by framework name?
If people start more than 1 marathon on the same role, we would fight each other.